PowerVCF.psm1

```diff
@@ -5690,6 +5690,618 @@ Export-ModuleMember -Function Get-VCFWsa
 #EndRegion APIs for managing Workspace ONE Access
 
 
+#Region APIs for managing Identity Providers
+
+Function Get-VCFIdentityProviders {
+    <#
+        .SYNOPSIS
+        Retrieves a list of all identity providers or the details of a specific identity provider.
+
+        .DESCRIPTION
+        The Get-VCFIdentityProviders cmdlet retrieves a list of all identity providers or the details of a specific
+        identity provider.
+
+        .EXAMPLE
+        Get-VCFIdentityProviders
+        This example shows how to retrieve the details of all identity providers.
+
+        .EXAMPLE
+        Get-VCFIdentityProviders -id 3e250ddd-07ec-4923-a161-ab6e9aa588181
+        This example shows how to retrieve the details of a specific identity provider.
+
+        .PARAMETER id
+        Specifies the unique ID of the identity provider.
+    #>
+
+    Param (
+        [Parameter (Mandatory = $false)] [ValidateNotNullOrEmpty()] [String]$id
+    )
+
+    Try {
+        $vcfVersion = Get-VCFManager -version
+        if ($vcfVersion -ge '4.5.0') {
+            createHeader # Set the Accept and Authorization headers.
+            checkVCFToken # Validate the access token and refresh, if necessary.
+            if ($PsBoundParameters.ContainsKey("id")) {
+                $uri = "https://$sddcManager/v1/identity-providers/$id"
+            } else {
+                $uri = "https://$sddcManager/v1/identity-providers"
+            }
+            $response = Invoke-RestMethod -Method GET -Uri $uri -Headers $headers
+            $response
```

Comment:
```suggestion
            if ($PsBoundParameters.ContainsKey("id")) {
                $uri = "https://$sddcManager/v1/identity-providers/$id"
                Invoke-RestMethod -Method GET -Uri $uri -Headers $headers
            } else {
                $uri = "https://$sddcManager/v1/identity-providers"
                (Invoke-RestMethod -Method GET -Uri $uri -Headers $headers).elements
            }
```

Reply:
Because the APIs return different information a) one is a list b) one is specific value we should show consistent output. When there is .elements in the API response we don't expose this to the user we simply display the list

This is another place where we can reduce code / variable usage as the output is not required to be in a local variable